src/uPlot.js

```diff
@@ -1737,22 +1746,37 @@ export default function uPlot(opts, data, then) {
 
 				if (drag.setScale) {
 					batch(() => {
-						if (drag.x) {
-							_setScale(xScaleKey,
-								scaleValueAtPos(select[LEFT], xScaleKey),
-								scaleValueAtPos(select[LEFT] + select[WIDTH], xScaleKey),
-							);
+						if (drag.x && !(src != null && !src.cursor.drag[xScaleKey])) {
```

Comment:
similar to comment above, unless src and self have different opts, we can just use the local `drag[xScaleKey]`. also, not loving the double-negatives here.

Reply:
it looks like `src != null` is serving the same purpose as `e == null` at the top of `mouseUp()` (to check if this is a synced event or a local event). in that case we should keep it uniform. i'd be okay switching the `e == null` condition since `src != null` may be a bit clearer.